app/controllers/products_controller.rb

```diff
@@ -0,0 +1,46 @@
+class ProductsController < ApplicationController
+  before_action :require_login, only: [:new, :create, :edit, :update, :destroy]
```

Comment:
It's a WIP and a reminder that we will need to require the user to be logged in for certain product actions.